shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -505,6 +507,43 @@ public void showSystemOverlays() {
     verify(fakeWindowInsetsController).show(WindowInsetsCompat.Type.navigationBars());
   }
 
+  @Config(sdk = 30)
+  @Test
+  public void verifyUpdateSystemUiOverlaysAppliesCurrentTheme() {
+    View fakeDecorView = mock(View.class);
+    Window fakeWindow = mock(Window.class);
+    when(fakeWindow.getDecorView()).thenReturn(fakeDecorView);
+    Activity fakeActivity = mock(Activity.class);
+    when(fakeActivity.getWindow()).thenReturn(fakeWindow);
+    PlatformChannel fakePlatformChannel = mock(PlatformChannel.class);
+    PlatformPlugin platformPlugin = new PlatformPlugin(fakeActivity, fakePlatformChannel);
+    WindowInsetsController fakeWindowInsetsController = mock(WindowInsetsController.class);
+    when(fakeWindow.getInsetsController()).thenReturn(fakeWindowInsetsController);
+
+    // Style that requires usage of all system bar APIs used in PlatformPlugin to update overlay
+    // style
+    SystemChromeStyle testStyle =
+        new SystemChromeStyle(
+            0XFF000000, Brightness.LIGHT, true, 0XFFC70039, Brightness.LIGHT, 0XFF006DB3, true);
+
+    platformPlugin.updateSystemUiOverlays();
+
+    verify(fakeWindow, never()).setStatusBarColor(anyInt());
+    verify(fakeWindow, never()).setNavigationBarColor(anyInt());
+    verify(fakeWindow, never()).setNavigationBarDividerColor(anyInt());
+    verify(fakeWindow, never()).setStatusBarContrastEnforced(anyBoolean());
+    verify(fakeWindow, never()).setNavigationBarContrastEnforced(anyBoolean());
+
+    platformPlugin.setSystemChromeSystemUIOverlayStyle(testStyle);
```

Comment:
if `updateSystemUiOverlays` already calls `setSystemChromeSystemUIOverlayStyle`, then this method could go back to being private. This also decreases the # of calls `times(1)`, which is expected, is that right?

Reply:
You're right, I can make it private again, but it will still be called two times because I'm calling to set the style and then to update it

Reply:
sg. thanks